File: problems/485/solution.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        maxOnes = curOnes = 0
        
        for i in range(len(nums)):
            if nums[i]:
                curOnes = curOnes + 1 if i > 0 and nums[i -1] else 1

            maxOnes = max(maxOnes, curOnes)
        
        return maxOnes

File: problems/485/solution.go
package main
func findMaxConsecutiveOnes(nums []int) int {
    curOnes, maxOnes := 0, 0
    
    for _, v := range nums {
        if v == 1 {
            curOnes++
        } else {
            curOnes = 0
        }
        
        maxOnes = max(maxOnes, curOnes)
    }
    
    return maxOnes
}

func max(a, b int) int {
    if a > b {
        return a
    }
    
    return b
}